From 5b6796eb8c43c6d16fcb92cc1c5cb4351c4e7050 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 25 May 2017 11:27:40 -0400
Subject: [PATCH 258/260] Fix one more coverity complaint

No idea why covscan thinks this is an "added" bug, since the file hasn't
changed in 3 years, but... yeah.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grub-core/normal/completion.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/grub-core/normal/completion.c b/grub-core/normal/completion.c
index 2c9b9e931..93aa0d8ed 100644
--- a/grub-core/normal/completion.c
+++ b/grub-core/normal/completion.c
@@ -284,7 +284,8 @@ complete_file (void)
 
       /* Cut away the filename part.  */
       dirfile = grub_strrchr (dir, '/');
-      dirfile[1] = '\0';
+      if (dirfile)
+	dirfile[1] = '\0';
 
       /* Iterate the directory.  */
       (fs->dir) (dev, dir, iterate_dir, NULL);
-- 
2.13.0

